jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1601,6 +1601,190 @@ ecma_builtin_typedarray_prototype_find_index (ecma_value_t this_arg, /**< this a
+                                              ecma_length_t args_number) /**< number of arguments */
+{
+  if (!ecma_is_typedarray (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a TypedArray."));
+  }
+  ecma_object_t *typedarray_p = ecma_get_object_from_value (this_arg);
+  uint32_t length = ecma_typedarray_get_length (typedarray_p);
+  uint32_t target = 0;
+  uint32_t start = 0;
+  uint32_t end = length;
+  ecma_number_t num_var;
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+
+  if (args_number > 0)
+  {
+    ret_value = ecma_get_number (args[0], &num_var);
+
+    if (ECMA_IS_VALUE_ERROR (ret_value))
+    {
+      return ret_value;
+    }
+
+    target = ecma_builtin_typedarray_prototype_copy_within_helper(length, num_var, false);
+    if (args_number > 1)
+    {
+      ret_value = ecma_get_number (args[1], &num_var);
+
+      if (ECMA_IS_VALUE_ERROR (ret_value))
+      {
+        return ret_value;
+      }
+
+      start = ecma_builtin_typedarray_prototype_copy_within_helper(length, num_var, false);
+      if (args_number > 2)
+      {
+        ret_value = ecma_get_number (args[2], &num_var);
+
+        if (ECMA_IS_VALUE_ERROR (ret_value))
+        {
+          return ret_value;
+        }
+
+        end = ecma_builtin_typedarray_prototype_copy_within_helper(length, num_var, true);
+      }
+    }
+
+  }
+
+  int32_t count = (int32_t) (end - start) > (int32_t) (length - target) ? (int32_t) (length - target) : (int32_t) (end - start);
+  int32_t direction;
+
+  if (start < target && target < start + (uint32_t) count)
+  {
+    direction = -1;
+    start += (uint32_t) count -1;
+    target += (uint32_t) count -1;
+  }
+  else
+  {
+    direction = 1;
+  }
+
+  while (count > 0)
+  {
+    ecma_value_t start_value = ecma_make_uint32_value (start);
+    ecma_value_t from_key = ecma_op_to_string (start_value);
+    if (ECMA_IS_VALUE_ERROR (from_key))
+    {
+      return from_key;
+    }
+    ecma_string_t *from_key_p = ecma_get_string_from_value (from_key);
+
+    ecma_value_t target_value = ecma_make_uint32_value (target);
+    ecma_value_t to_key = ecma_op_to_string (target_value);
+    if (ECMA_IS_VALUE_ERROR (to_key))
+    {
+      return to_key;
+    }
+    ecma_string_t *to_key_p = ecma_get_string_from_value (to_key);
+
+    ecma_value_t from_present = ecma_op_object_find (typedarray_p, from_key_p);
+
+    if (from_present != ECMA_VALUE_NOT_FOUND)
+    {
+      ecma_value_t from_val = ecma_op_object_get (typedarray_p, from_key_p);
+      ecma_value_t set_status = ecma_op_object_put (typedarray_p, to_key_p, from_val, true);
+
+      if (ECMA_IS_VALUE_ERROR (set_status))
+      {
+        return set_status;
+      }
+
+      ecma_free_value (from_val);
+      ecma_free_value (set_status);
+    }
+    else
+    {
+      ecma_value_t delete_status = ecma_op_object_delete (typedarray_p, to_key_p, true);
+
+      if (ECMA_IS_VALUE_ERROR (delete_status))
+      {
+        return delete_status;
```

Comment:
If we return with error here (and above) a lot of values are not freed (`from_key_p`, `to_key_p`, etc...)